### PR TITLE
KYLIN-4033 Can not access Kerberized Cluster with DebugTomcat

### PR DESCRIPTION
**Pull request by TempleZhou**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: server/src/main/java/org/apache/kylin/rest/DebugTomcat.java [hunk 1]
```diff
@@ -99,6 +107,25 @@ private static File findFile(String dir, String ptn) {
         }
         return null;
     }
+    public static boolean isKerberosEnabled() {
+        Configuration conf = new Configuration();
+        boolean authEnabled = conf.getBoolean("hadoop.security.authorization", false);
+        String authMethod = conf.get("hadoop.security.authentication");
+        return authEnabled && authMethod.toLowerCase(Locale.ROOT).equals("kerberos");
```
Comment (hn5092) [inline server/src/main/java/org/apache/kylin/rest/DebugTomcat.java]: UserGroupInformation.isSecurityEnabled()

File: server/src/main/java/org/apache/kylin/rest/DebugTomcat.java [hunk 2]
```diff
@@ -99,6 +107,25 @@ private static File findFile(String dir, String ptn) {
...
+    }
+
+    public static void authKrb5() {
+        System.setProperty("java.security.krb5.conf", "/etc/krb5.conf");
+        Configuration conf = new Configuration();
+        UserGroupInformation.setConfiguration(conf);
```
Comment (hn5092) [inline server/src/main/java/org/apache/kylin/rest/DebugTomcat.java]: UGI needn't set hadoopconf.org.apache.hadoop.security.UserGroupInformation#isSecurityEnabled will init it.

File: server/src/main/java/org/apache/kylin/rest/DebugTomcat.java [hunk 3]
```diff
@@ -99,6 +107,25 @@ private static File findFile(String dir, String ptn) {
...
+        try {
+            UserGroupInformation.loginUserFromKeytab(
```
Comment (hn5092) [inline server/src/main/java/org/apache/kylin/rest/DebugTomcat.java]: Use another way? Systemprop or env and note to dev doc.

File: server/src/main/java/org/apache/kylin/rest/DebugTomcat.java [hunk 4]
```diff
@@ -99,6 +107,25 @@ private static File findFile(String dir, String ptn) {
...
     }
+    public static boolean isKerberosEnabled() {
+        Configuration conf = new Configuration();
+        boolean authEnabled = conf.getBoolean("hadoop.security.authorization", false);
+        String authMethod = conf.get("hadoop.security.authentication");
+        return authEnabled && authMethod.toLowerCase(Locale.ROOT).equals("kerberos");
+    }
+
+    public static void authKrb5() {
+        System.setProperty("java.security.krb5.conf", "/etc/krb5.conf");
```
Comment (hn5092) [inline server/src/main/java/org/apache/kylin/rest/DebugTomcat.java]: System prop should note to dev doc.

Comment (TempleZhou) [reply server/src/main/java/org/apache/kylin/rest/DebugTomcat.java]: You are right. And, maybe, the keytab principal and path should be set using System.setProperty or Kylin Config. Do you have any idea?

File: server/src/main/java/org/apache/kylin/rest/DebugTomcat.java (hunk 2 above)
Comment (TempleZhou) [reply server/src/main/java/org/apache/kylin/rest/DebugTomcat.java]: Awesome!

**Pull request merged**